pkg/virt-controller/services/template.go

```diff
@@ -446,12 +446,31 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 			})
 		}
 		if volume.DataVolume != nil {
-			volumeMounts = append(volumeMounts, volumeMount)
+			logger := log.DefaultLogger()
+			claimName := volume.DataVolume.Name
```

Comment:
In CDI it is a basic assumption, I think it is OK to rely on it.

Reply:
Looking at the DataVolume spec:
```yaml
- apiVersion: cdi.kubevirt.io/v1alpha1
  kind: DataVolume
  ...
  spec:
    pvc:
      accessModes:
      - ReadWriteMany
      dataSource: null
      resources:
        requests:
          storage: 1Gi
      storageClassName: csi-rbd
      volumeMode: Block
    source:
    ...
```
The PVC inside doesn't seem to include its identifier (name), right?
So yeah, I agree it will require querying the corresponding PVC but I think the query would need to be based on the `OwnerReference` within the PVC:
```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
...
  ownerReferences:
  - apiVersion: cdi.kubevirt.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: DataVolume
    name: my-datavolume
    uid: 38703a07-c95f-11e9-9e6c-525500d15501
```

But anyway, I'm not suggesting to do it now. I just argue that we may end up changing CDI at some point to generate an arbitrary name for the generated PVC, like XCVASDA34SD :) And once we do that, having errors like `didn't find PVC associated with DataVolume: my-great-dvv` would be better for debugging than errors like `didn't find PVC XCVASDA34SD`. Hence the motivation for keeping code paths for DataVolumes separate than those of PVCs rather than unifying them.

Reply:
PVC spec on the DV includes the block mode setting (volumeMode: Block). Shouldn't need a lookup of actual PVC again. I'll leave it up to you decide if its worth doing it or not.

Reply:
Yeah, querying the DV may also be a possible solution. I propose to keep that open at this point, we don't have to decide it now

Reply:
We don't have the DV available at this point, so it doesn't matter if we query the DV or PVC. And the PVC lookup func is already there. So I think it's fine to do it this way.